libs/SmartStore/src/com/salesforce/androidsdk/smartstore/store/SmartStore.java

```diff
@@ -241,6 +241,8 @@ public JSONObject getLastExplainQueryPlan() {
      * Get database size
      */
     public int getDatabaseSize() {
+		// With WAL enabled we must force a WAL checkpoint if we want the actual DB file size.
+		getDatabase().query("PRAGMA wal_checkpoint(FULL);").moveToNext();
     	int size =  (int) (new File(getDatabase().getPath()).length()); // XXX That cast will be trouble if the file is more than 2GB
```

Comment:
Side note:  The 2GB file size limit was artificially imposed by Android back in the day but I believe it was lifted long ago to the real (FAT32) hard limit of 4GB.  Does anybody know when?  And is this actually a dangerous cast?

Reply:
`int` is signed, so it only goes up to +2^31 anyway.  While the cast may succeed, the size may be negative in those cases where it is bigger than ~2GB

Reply:
Whoops, I completely missed that. Thanks.